core/templates/hashfuncs.h

```diff
@@ -413,6 +420,13 @@ struct HashMapComparatorDefault<double> {
 	}
 };
 
+template <>
+struct HashMapComparatorDefault<Color> {
+	static bool compare(const Color &p_lhs, const Color &p_rhs) {
+		return ((p_lhs.r == p_rhs.r) || (Math::is_nan(p_lhs.r) && Math::is_nan(p_rhs.r))) && ((p_lhs.g == p_rhs.g) || (Math::is_nan(p_lhs.g) && Math::is_nan(p_rhs.g))) && ((p_lhs.b == p_rhs.b) || (Math::is_nan(p_lhs.b) && Math::is_nan(p_rhs.b))) && ((p_lhs.a == p_rhs.a) || (Math::is_nan(p_lhs.a) && Math::is_nan(p_rhs.a)));
```

Comment:
```suggestion
		return HashMapComparatorDefault<float>().compare(p_lhs.r, p_rhs.r) && HashMapComparatorDefault<float>().compare(p_lhs.g, p_rhs.g) && HashMapComparatorDefault<float>().compare(p_lhs.b, p_rhs.b) && HashMapComparatorDefault<float>().compare(p_lhs.a, p_rhs.a);
```
For the general, as a more clean solution with less risk of errors, can add if desired